lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -41,17 +41,13 @@ def initialize(gem, indent = 0, include_doc = false)
           gem.parse_yard_docs if include_doc
         end
 
-        sig { returns(String) }
+        sig { returns(RBI::Tree) }
         def generate
```

Comment:
Can we change the signature of this method so that it takes an `RBI::Tree` instance (that it then updates and returns)? If we do that, then in the outer generator level we can instantiate an `RBI::File` as early as possible with the provided strictness level, etc and then pass in its root tree here to be modified.

Moreover, an extension method on `RBI::File` in Tapioca can be responsible for doing the nesting and grouping and sorting before calling the actual `RBI::File#string` method, like what @Morriar is asking for below.

Reply:
I'm not 100% sure I get what you're asking for. You want the generate to accept an `RBI::Tree` and return an `RBI::Tree`?

Reply:
Yes, exactly. My idea is for code to flow like something like this:
```ruby
rbi_file = RBI::File.new(strictness: "true")
SymbolTableCompiler.new.generate(rbi_file.root)
File.write("foo.rbi", rbi_file.to_s) # where `to_s` calls `transform_rbi` on the `root` internally
```
We can then make file operations be another abstraction later on:
```
rbi_file = Tapioca:RBIFile.new(filename, strictness: "true")
SymbolTableCompiler.new.generate(rbi_file.root)
rbi_file.save
```

Reply:
Perfect. That makes much more sense. Will start on implementing that now. 👍🏻